sdk/package/sdk.ts

```diff
@@ -0,0 +1,726 @@
+        console.log("post" + i + ":" + JSON.stringify(posts[i]));
+        const convertedPost = transfer_post(posts[i]);
+        console.log("convert post:" + JSON.stringify(convertedPost));
+        const transaction = await mapPostToTransaction(convertedPost, api);
+        console.log("transaction:" + JSON.stringify(transaction));
+        transactions.push(transaction);
+    }
+    const txs = await transactionsToBatches(transactions, api);
+    return {
+        posts,
+        transactions,
+        txs
+    }
+}
+
+/// Using sign on wallet and using signdAndSend to polkadot.js transaction
+const sign_and_send = async (api: ApiPromise, signer: string, wasm: any, wasmWallet: Wallet, assetMetadataJson: any, transaction: Transaction, network: Network): Promise<void> => {
+
+    const signed_transaction = await sign_transaction(api,wasm,wasmWallet,assetMetadataJson,transaction,network);
+
+    for (let i = 0; i < signed_transaction.txs.length; i++) {
+        try {
+            await signed_transaction.txs[i].signAndSend(signer, (_status, _events) => { });
+        } catch (error) {
+            console.error('Transaction failed', error);
+        }
+    }
+}
+
+/// Maps a given `post` to a known transaction type, either Mint, Private Transfer or Reclaim.
+async function mapPostToTransaction(post: any, api: ApiPromise): Promise<SubmittableExtrinsic<"promise", any>> {
+    let sources = post.sources.length;
+    let senders = post.sender_posts.length;
+    let receivers = post.receiver_posts.length;
+    let sinks = post.sinks.length;
+
+    if (sources == 1 && senders == 0 && receivers == 1 && sinks == 0) {
+        const mint_tx = await api.tx.mantaPay.toPrivate(post);
+        return mint_tx;
+    } else if (sources == 0 && senders == 2 && receivers == 2 && sinks == 0) {
+        const private_transfer_tx = await api.tx.mantaPay.privateTransfer(post);
+        return private_transfer_tx;
+    } else if (sources == 0 && senders == 2 && receivers == 1 && sinks == 1) {
+        const reclaim_tx = await api.tx.mantaPay.toPublic(post);
+        return reclaim_tx;
+    } else {
+        throw new Error(
+            'Invalid transaction shape; there is no extrinsic for a transaction'
+            + `with ${sources} sources, ${senders} senders, `
+            + ` ${receivers} receivers and ${sinks} sinks`
+        );
+    }
+};
+
+/// Batches transactions.
+async function transactionsToBatches(transactions: any, api: ApiPromise): Promise<SubmittableExtrinsic<"promise", any>[]> {
+    const MAX_BATCH = 2;
+    const batches = [];
+    for (let i = 0; i < transactions.length; i += MAX_BATCH) {
+        const transactionsInSameBatch = transactions.slice(i, i + MAX_BATCH);
+        const batchTransaction = await api.tx.utility.batch(
+            transactionsInSameBatch
+        );
+        batches.push(batchTransaction);
+    }
+    return batches;
+}
+
+/// NOTE: `post` from manta-rs sign result should match runtime side data structure type.
```

Comment:
So runtime ought to catch up with manta-rs types, right? Is there a github issue for this?

Reply:
manta-rs has its type definition, and runtime also has its own type definition that using raw byte array which maybe easier on node side. We can catch up the manta-rs types on runtime side in later PR. But for now, I think we can just doing convert type transformation on sdk.